readerwriterqueue.h

```diff
@@ -51,8 +51,8 @@
 #ifndef MOODYCAMEL_MAYBE_ALIGN_TO_CACHELINE
 #if defined (__APPLE__) && defined (__MACH__) && __cplusplus >= 201703L
 // This is required to find out what deployment target we are using
-#include <CoreFoundation/CoreFoundation.h>
-#if !defined(MAC_OS_X_VERSION_MIN_REQUIRED) || MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_14
+#include <AvailabilityMacros.h>
+#if !defined(MAC_OS_X_VERSION_MIN_REQUIRED) || (defined(MAC_OS_X_VERSION_10_14) && MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_14)
```

Comment:
In that case, shouldn't this be
```suggestion
#if !defined(MAC_OS_X_VERSION_MIN_REQUIRED) || !defined(MAC_OS_X_VERSION_10_14) || MAC_OS_X_VERSION_MIN_REQUIRED < MAC_OS_X_VERSION_10_14
```

Reply:
Looks good to me, I'll change it!